### ISSUE TITLE
Stage pop up always jumps to top

### ISSUE THREAD

**Issue by mangelozzi** [NeogitOrg/neogit#841]
### Description

Great plugin, but I often regret updating it. I can't use the stage popup anymore, because I have a lot of changes, and every time I press tab, its scrolls back to the top of the screen, super disorientating.

### Neovim version

NVIM v0.10.0-dev-fcfc87c
Build type: RelWithDebInfo
LuaJIT 2.1.1695266395
Run "nvim -V1 -v" for more info

### Operating system and version

Ubuntu 22.04

### Steps to reproduce

Open the commit pop up,
Scroll down to a diff, press tab to open, the cusor and screen will scroll to the top and you will loose your place.
Also happens is you press `s` to stage a section, scrolls back to the top of the screen

### Expected behavior

Cursor and screen position to stay where it was

### Actual behavior

Jumps back to top 

### Minimal config

```Lua
vim.cmd([[set runtimepath=$VIMRUNTIME]])
vim.cmd([[set packpath=/tmp/nvim/site]])
local package_root = "/tmp/nvim/site/pack"
local install_path = package_root .. "/packer/start/packer.nvim"
local function load_plugins()
  require("packer").startup({
    {
      "wbthomason/packer.nvim",
      {
        "NeogitOrg/neogit",
        requires = {
          { "nvim-lua/plenary.nvim" },
          { "sindrets/diffview.nvim" },
        },
        config = function()
          print("loaded neogit")
          require("neogit").setup()
        end,
      },
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. "/plugin/packer_compiled.lua",
      display = { non_interactive = true },
    },
  })
end
if vim.fn.isdirectory(install_path) == 0 then
  print("Installing neogit and dependencies.")
  vim.fn.system({ "git", "clone", "--depth=1", "https://github.com/wbthomason/packer.nvim", install_path })
end
load_plugins()
require("packer").sync()
```

**Comment by CKolkey** [comment]
Yeah, theres a double-refresh bug I havent had time to sort out yet. Try turning off the file watcher, that should do the trick 😘

**Comment by CKolkey** [comment]
I've been thinking about this: It's very annoying that the cursor jumps, so until I sort out the double-render thing, I'm disabling the filewatcher by default (you can opt in).

Thank you for the feedback :) I'm closing this issue, but please feel free to reopen it if you continue to experience this bug.

**Comment by mangelozzi** [comment]
Thanks for tending to this, but it still jump around for me: Just so we on the same page, here's a screen shot below.
I navigate to `urls.py` and press and expand the file, you can see although the cursor was at the top of the screen so that when it expands, I would be able to read it, but it moves the current line to the bottom of the screen, and expands it such that one can only see 1 or 2 lines. Then I try to expand `views.py` and they same thing, jumps to the bottom of the screen. The craziness after that is me trying to move right with `l`.

![neogitcraziness](https://github.com/NeogitOrg/neogit/assets/19764314/78ab2270-22a4-45ec-81bf-d30564b826f1)

**Comment by AlexanderArvidsson** [comment]
This is also happening for me. Every time I expand a file, it scrolls to the top, which makes is super hard to quickly go through lots of files changes.

Please re-open!

**Comment by CKolkey** [comment]
This should no longer be an issue on the `nightly` branch.

**Comment by AlexanderArvidsson** [comment]
@CKolkey  I've been experiencing similar issues again on the `nightly` branch, and I've started looking at the code. It doesn't exactly jump to the top, but the scroll jumps.

It seems like the way the cursor is stored and restored on `Ui:update` needs some love, because while the cursor _does_ restore correctly, the _top line_ does not (the scroll). When updating the UI, the cursor is moved to the absolute top due to the `self.buf:clear()`, and then restores the cursor position. The problem is, that counts as moving the cursor down to the restored line, so the scroll will "fit" to the cursor just as if you'd do "zb".

I've already made some minor adjustments to save and restore the top line and will make a PR soon!
